modules/consensus/tendermint/prover/src/client.rs

```diff
@@ -213,7 +225,10 @@ impl Client for CometBFTClient {
 /// Heimdall nodes provide a JSON-RPC interface for querying blockchain data.
 pub struct HeimdallClient {
 	raw_client: ReqwestClient,
-	base_url: String,
+	consensus_rpc_url: String,
+	http_client: HttpClient,
+	rest_url: String,
+	execution_rpc_client: Arc<Provider<Http>>,
```

Comment:
Also not needed for tendermint chains. Maybe create a new folder called polygon in `modules/consensus` that wraps all this tendermint stuff

Reply:
This is unresolved